CoreVulkanDisplay/src/au/gov/asd/tac/constellation/visual/vulkan/renderables/CVKIconsRenderable.java

```diff
@@ -84,28 +166,81 @@
+    private CVKRenderableResourceState vertexBuffersState = CVK_RESOURCE_CLEAN;
+    private CVKRenderableResourceState commandBuffersState = CVK_RESOURCE_CLEAN;
+    private CVKRenderableResourceState descriptorSetsState = CVK_RESOURCE_CLEAN;
+    private CVKRenderableResourceState pipelinesState = CVK_RESOURCE_CLEAN;    
+    private long hAtlasSampler = VK_NULL_HANDLE;
+    private long hAtlasImageView = VK_NULL_HANDLE;    
+        
+    // Resources recreated with the swap chain (dependent on the image count)    
+    private LongBuffer pDescriptorSets = null; 
+    private List<Long> pipelines = null;
+    private List<CVKCommandBuffer> commandBuffers = null;    
+    private List<CVKBuffer> vertexBuffers = null;   
+    private List<CVKBuffer> vertexUniformBuffers = null;
+    private List<CVKBuffer> geometryUniformBuffers = null;    
+    private List<CVKBuffer> fragmentUniformBuffers = null;        
     
-    // TODO_TT: not really width....
-    private static final int ICON_BUFFER_WIDTH = 4;
-    private static final int COLOR_BUFFER_WIDTH = 4;
-    private static final int XYZ_BUFFER_WIDTH = 8;
+    // The UBO staging buffers are a know size so created outside user events
+    private CVKBuffer cvkVertexUBStagingBuffer = null;
+    private CVKBuffer cvkGeometryUBStagingBuffer = null;
+    private CVKBuffer cvkFragmentUBStagingBuffer = null;
+    private VertexUniformBufferObject vertexUBO = new VertexUniformBufferObject();
+    private GeometryUniformBufferObject geometryUBO = new GeometryUniformBufferObject();
+    private FragmentUniformBufferObject fragmentUBO = new FragmentUniformBufferObject();    
     
-
+    // Resources recreated only through user events
     private int vertexCount = 0;
     private CVKBuffer cvkVertexStagingBuffer = null;
-    private CVKBuffer cvkXYZWStagingBuffer = null;
-    private CVKBuffer cvkXYZWTexelBuffer = null;
-    private long hXYZWBufferView = VK_NULL_HANDLE;
-    private List<Long> pipelines = null;
-    private List<Long> pipelineLayouts = null;    
-    private List<CVKBuffer> vertexBuffers = null;    
-    private boolean recreateIcons = false;
-    private ReentrantLock vertexLock = new ReentrantLock();
+    private CVKBuffer cvkPositionStagingBuffer = null;
+    private CVKBuffer cvkPositionBuffer = null;    
+    private CVKBuffer cvkVertexFlagsStagingBuffer = null;
+    private CVKBuffer cvkVertexFlagsBuffer = null;    
+    private long hPositionBufferView = VK_NULL_HANDLE;
+    private long hVertexFlagsBufferView = VK_NULL_HANDLE;
+    
+    // The vertex, position and flags staging buffers are used by both the event
+    // thread and rendering thread so much be synchronised.
+    private ReentrantLock vertexStagingBufferLock = new ReentrantLock();
+    private ReentrantLock positionStagingBufferLock = new ReentrantLock();
+    private ReentrantLock vertexFlagsStagingBufferLock = new ReentrantLock();
+    
+    
+    /*
```

Comment:
change comment block style

Reply:
good comment though

Reply:
Thanks, I moved it to the class Javadoc.  It will need more fleshing out later.  We need to do a doco pass at some point.  Probably before we reduce our load on the project so we can better re-acclimate when returning to it.